effekt/shared/src/main/scala/effekt/core/Tree.scala

```diff
@@ -59,39 +74,44 @@ case class ModuleDecl(
   declarations: List[Declaration],
   externs: List[Extern],
   definitions: List[Definition],
-  exports: List[Symbol]
+  exports: List[Id]
 ) extends Tree
 
 /**
  * Toplevel data and interface declarations
  */
 enum Declaration extends Tree {
-  def id: Symbol
+  def id: Id
 
-  case Data(id: Symbol, ctors: List[Symbol])
-  case Record(id: Symbol, fields: List[Symbol])
-  case Interface(id: Symbol, operations: List[Symbol])
+  case Data(id: Id, tparams: List[Id], constructors: List[Constructor])
+  case Record(id: Id, tparams: List[Id], constructor: Constructor)
```

Comment:
Consider absorbing `Record` into `Data` (and likewise on the object language type level). Keep `Select` (it happens to work on all data types with a single constructor).

Reply:
I agree, let's do this. However, we should do this after this PR since we also need to push this change through lifted and I don't want to touch lifted too much in this PR.